open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +272,429 @@ public static JSONArray appendDummyAccountID() {
+
+        //Adding Payment Type
+        paymentTypeArray.add(ConsentExtensionConstants.DOMESTIC_VRP);
+        jsonElementPaymentType.appendField(ConsentExtensionConstants.TITLE,
+                ConsentExtensionConstants.PAYMENT_TYPE_TITLE);
+        jsonElementPaymentType.appendField(ConsentExtensionConstants.DATA_SIMPLE, paymentTypeArray);
+        consentDataJSON.add(jsonElementPaymentType);
+
+        // Constructing jsonElementValidToDataTime
+        JSONObject jsonElementValidToDateTime = new JSONObject();
+        jsonElementValidToDateTime.appendField(ConsentExtensionConstants.TITLE,
+                ConsentExtensionConstants.CONTROL_PARAMETER_VALID_TO_DATE_TITLE);
+        JSONArray dateControlParameterArray = new JSONArray();
+        dateControlParameterArray.add((controlParameters).
+                get(ConsentExtensionConstants.VALID_TO_DATE_TIME));
+        jsonElementValidToDateTime.appendField(ConsentExtensionConstants.DATA_SIMPLE, dateControlParameterArray);
+
+        // Constructing jsonElementValidFromDataTime
+        JSONObject jsonElementValidFromDateTime = new JSONObject();
+        jsonElementValidFromDateTime.appendField(ConsentExtensionConstants.TITLE,
+                ConsentExtensionConstants.CONTROL_PARAMETER_VALID_FROM_DATE_TITLE);
+        JSONArray dateTimeControlParameterArray = new JSONArray();
+        dateTimeControlParameterArray.add((controlParameters).
+                get(ConsentExtensionConstants.VALID_FROM_DATE_TIME));
+        jsonElementValidFromDateTime.appendField(ConsentExtensionConstants.DATA_SIMPLE, dateTimeControlParameterArray);
+
+        // Constructing jsonElementControlParameter
+        JSONObject jsonElementControlParameter = new JSONObject();
+        jsonElementControlParameter.appendField(ConsentExtensionConstants.TITLE,
+                ConsentExtensionConstants.CONTROL_PARAMETER_MAX_INDIVIDUAL_AMOUNT_TITLE);
+        JSONArray controlParameterArray = new JSONArray();
+
+        String formattedAmount = String.format("%s %s",
+                ((JSONObject) controlParameters.get(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT))
+                        .get(ConsentExtensionConstants.CURRENCY),
+                ((JSONObject) controlParameters.get(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT))
+                        .get(ConsentExtensionConstants.AMOUNT));
+        controlParameterArray.add(formattedAmount);
+        jsonElementControlParameter.appendField(ConsentExtensionConstants.DATA_SIMPLE, controlParameterArray);
+
+        // Constructing jsonElementPeriodAlignment
+        JSONObject jsonElementPeriodAlignment = new JSONObject();
+        jsonElementPeriodAlignment.appendField(ConsentExtensionConstants.TITLE,
+                ConsentExtensionConstants.CONTROL_PARAMETER_PERIOD_ALIGNMENT_TITLE);
+        String periodAlignment = (String) ((JSONObject) ((JSONArray) controlParameters
+                .get(ConsentExtensionConstants.PERIODIC_LIMITS)).get(0))
```

Comment:
Can there be null pointers in the statement

Reply:
Fixed in [a6d2b0b]